core-services/ipi-deprovision/clusterpool-pruner.sh

```
@@ -54,15 +54,27 @@ oc --context hosted-mgmt get clusterpool -A -o jsonpath='{range .items[*]}{.meta
+    echo "  Patching ${cd} preserveOnDelete=false"
+    oc --context hosted-mgmt -n "$ns" patch "$cd" --type=merge -p '{"spec":{"preserveOnDelete":false}}'
+    echo "  Deleting ${cd}"
+    oc --context hosted-mgmt -n "$ns" delete "$cd" --wait=false
+  done < <(oc --context hosted-mgmt -n "$ns" get clusterdeployment -o name 2>/dev/null || true)
+  oc --context hosted-mgmt delete clusterpool -n "$ns" "$name" --wait=false
+}
+
 if [[ -s /tmp/clusterpools.stale ]]; then
   stale_count=$(wc -l < /tmp/clusterpools.stale)
   echo "Found $stale_count zombie clusterpool(s) to clean up:"
   while read ns name; do
```

Shellcheck (0.11.0)

[info] read without -r will mangle backslashes.

(SC2162)